gnocchi/carbonara.py

```diff
@@ -490,6 +490,11 @@ def __hash__(self):
 
     def _compare(self, op, other):
         if isinstance(other, SplitKey):
+            if op == operator.eq:
+                return (op(self.key, other.key) and
+                        op(self.sampling, other.sampling))
+            if op == operator.ne:
+                return not self._compare(operator.eq, other)
             if self.sampling != other.sampling:
```

Comment:
can't you just check if op is operator.eq or op is operator.ne and return False as i believe the only scenario is if sampling is different it's automatically not equivalent.

Reply:
I don't think so IIUC since the final call is `op(self.key, other.key)` and does not include the sampling in comparison then.

Reply:
That'd (almost) work too. Updating the PR.